pkg/vault/contracts/Vault.sol

```diff
@@ -1159,16 +1154,19 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
             emit Wrap(params.wrappedToken, amountInRaw, amountOutRaw, bufferBalances);
         }
 
-        if (params.kind == SwapKind.EXACT_IN) {
-            if (amountOutRaw < params.limitRaw) {
-                revert SwapLimit(amountOutRaw, params.limitRaw);
-            }
-            amountCalculatedRaw = amountOutRaw;
-        } else {
-            if (amountInRaw > params.limitRaw) {
-                revert SwapLimit(amountInRaw, params.limitRaw);
-            }
-            amountCalculatedRaw = amountInRaw;
+        _ensureValidWrapAmount(params.wrappedToken, amountCalculatedRaw);
+
+        if (amountCalculatedRaw < params.limitRaw) {
+            revert SwapLimit(amountCalculatedRaw, params.limitRaw);
+        }
```

Comment:
We still need this `if / else` unfortunately. For exact in, we check `amountCalculatedRaw < limit`, but for exact out we check `amountCalculatedRaw > limiti`.